container-search/src/main/java/com/yahoo/search/ranking/FunEvalCtx.java

```diff
@@ -0,0 +1,7 @@
+// Copyright Yahoo. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.search.ranking;
+
+import java.util.List;
+import java.util.function.Supplier;
+
+record FunEvalCtx(Supplier<Evaluator> evalSrc, List<String> fromQuery, List<String> fromMF) {}
```

Comment:
consider Spec (or similar) instead of Ctx, to indicate a plan/config rather than per-something state